community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/ExecutionEngineTests.java

```diff
@@ -48,11 +55,18 @@
     @Test
     public void shouldConvertListsAndMapsWhenPassingFromScalaToJava() throws Exception
     {
-        GraphDatabaseCypherService graph = new GraphDatabaseCypherService( this.database.getGraphDatabaseAPI() );
+        GraphDatabaseQueryService graph = new GraphDatabaseCypherService( this.database.getGraphDatabaseAPI() );
         ExecutionEngine executionEngine = new ExecutionEngine( graph, NullLogProvider.getInstance() );
 
-        Result result = executionEngine.executeQuery(
-                "RETURN { key : 'Value' , collectionKey: [{ inner: 'Map1' }, { inner: 'Map2' }]}", NO_PARAMS, SESSION );
+
+        Result result;
+        try ( InternalTransaction tx = graph.beginTransaction( KernelTransaction.Type.implicit, AccessMode.FULL ) )
+        {
+            result = executionEngine.executeQuery(
+                    "RETURN { key : 'Value' , collectionKey: [{ inner: 'Map1' }, { inner: 'Map2' }]}", NO_PARAMS,
+                    newSession( graph, tx ) );
+            tx.success();
+        }
```

Comment:
So, I know we're not supposed to use `ExecutionEngine` directly to execute Cypher anymore, but if we were, is this now the recommended pattern to use it in?